src/network/networkd-ndisc.c

```diff
@@ -389,6 +389,35 @@ static int ndisc_router_process_icmp6_ratelimit(Link *link, sd_ndisc_router *rt)
         return 0;
 }
 
+static int ndisc_router_process_retrans_time(Link *link, sd_ndisc_router *rt) {
+        char buf[DECIMAL_STR_MAX(usec_t)];
+        usec_t retrans_time;
+        int r;
+
+        assert(link);
+        assert(link->network);
+        assert(rt);
+
+        if (!link->network->ipv6_accept_ra_use_retrans_time)
+                return 0;
+
+        r = sd_ndisc_router_get_retrans_time(rt, &retrans_time);
+        if (r < 0) {
+                log_link_debug(link, "Failed to get ICMP6 ratelimit from RA, ignoring: %m");
+                return 0;
+        }
+
+        /* Set the retransmission time for Neigbor Solicitations.
+         * 0 is the unspecified value and must not be set (see RFC4861, 6.3.4) */
+        if (retrans_time != 0) {
+                xsprintf(buf, USEC_FMT, DIV_ROUND_UP(retrans_time, USEC_PER_MSEC));
+                r = sysctl_write_ip_neigh_property(AF_INET6, link->ifname, "retrans_time_ms", buf);
```

Comment:
Why not to use `sysctl_write_ip_neigh_property_uint32()` ?

Reply:
I could have, though I opted for consistency with the algorithm used in ndisc_router_process_icmp6_ratelimit().

Reply:
Then, please use. And let's update ndisc_router_process_icmp6_ratelimit().

Reply:
Will do.

Reply:
Done.

Reply:
-> #30898.